comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -233,6 +276,126 @@ func (c *collector) parseTaskContainers(
 	return taskContainers, events
 }
 
+func (c *collector) parseV4Task(ctx context.Context, task v1.Task, seen map[workloadmeta.EntityID]struct{}) []workloadmeta.CollectorEvent {
+	v4Task := c.getV4TaskWithTags(ctx, task)
+	return util.ParseV4Task(v4Task, seen)
+}
```

Comment:
I find the function to provide some indirection when reading the code. 

When I was looking at the code from line 166, I was thinking why do we have two function named the same but one being public and the other being private. 

Then I realized that `parseV4Task` is ultimately called `ParseV4Task`.

What do you think of getting rid of this private function and inline in line 169? 

```go
if _, ok := taskArns[task.Arn]; !ok {
	events = append(events, util.ParseV4Task(v1TaskToV4Task(task), seen)...)
} else {
	v4Task := c.getV4TaskWithTags(ctx, task)
	events = append(events, util.ParseV4Task(v4Task, seen)...)
}
```

What do you think?

Reply:
https://github.com/DataDog/datadog-agent/blob/882d6e4354906f23c0124aa0d908eed0e5bcbd4a/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L215-L222